frontend/app/controllers/Joiner.scala

```diff
@@ -170,11 +183,24 @@ trait Joiner extends Controller with ActivityTracking {
       Future.sequence(optFuture.toSeq).map(_.headOption)
     }
 
+    val futureContentOpt = request.cookies.get("GU_MEM_REFERER").map(_.value).map { referer =>
+      val refererPath = referer.replace("http://www.theguardian.com", "")
+      contentApiService.contentItemQuery(refererPath).map(_.content.map(MembersOnlyContent))
+    }.getOrElse(Future.successful(None))
+
```

Comment:
I had a look at this again after destroying your attempt this morning..sorry about that.

What you have written works but I'd split it up a bit. Currently the code is mixing the working out what the item path is and calling the Content API with the path.

So I would do something like:

```
val refererCookie = request.cookies.get("GU_MEM_REFERER").map(_.value.replace("http://www.theguardian.com", ""))
    
val futureContentOpt = refererCookie.map { referer =>
  contentApiService.contentItemQuery(referer).map(_.content.map(MembersOnlyContent))
}.getOrElse(Future.successful(None))
```